### PR TITLE
feat(layout): detect screen size

### PR DESCRIPTION
**Pull request by dryabinin-actionengine**

### CODE REVIEW

File: src/utils/layout.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  };
+};
+
+/**
+ * Return properties based on current layout
+ * @param properties 
+ */
+export const getCurrentLayoutProperty = (properties: LayoutProperties) => (props: any): string | number => {
+  const layoutObject = props?.layout || DEFAULT_LAYOUT;
+  const detectedLayoutKey = Object.entries(layoutObject).find(([_, value]) => value)?.[0] || DEFAULT_LAYOUT_KEY;
```
Comment (belom88) [inline src/utils/layout.ts:29]: fix warining

File: src/utils/layout.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import type { Layout, LayoutProperties } from "./types";
+
+import { useMediaQuery } from "react-responsive";
+
+const DEFAULT_LAYOUT = { isMobile: true };
```
Comment (belom88) [inline src/utils/layout.ts:5]: why mobile if default?

File: src/utils/layout.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+const DEFAULT_LAYOUT_KEY = 'isMobile';
+
+/**
+ * Detect current device
+ */
+export const useAppLayout = (): Layout => {
+  const isDesktopOrLaptop = useMediaQuery({ minWidth: 992 });
```
Comment (belom88) [inline src/utils/layout.ts:12]: Is it possible that there are >1 layout types are `true`. If not, I don't see the reason to return `true false false` or `false true false` or `false false true` map. It is excess data.

**Review by belom88 — CHANGES_REQUESTED** [review]

File: src/utils/layout.ts (hunk 1 above)
Comment (dryabinin-actionengine) [reply src/utils/layout.ts]: Done

File: src/utils/layout.ts (hunk 2 above)
Comment (dryabinin-actionengine) [reply src/utils/layout.ts]: Changed

File: src/utils/layout.ts (hunk 3 above)
Comment (dryabinin-actionengine) [reply src/utils/layout.ts:10]: Yes I see, I have simplified logic.

**Review by belom88 — APPROVED** [review]

**Pull request merged**